cocos/audio/apple/AudioPlayer.mm

```diff
@@ -125,10 +132,19 @@ of this software and associated documentation files (the "Software"), to deal
         }
     } while(false);
     
-    ALOGVV("Before alSourceStop");
-    alSourceStop(_alSource); CHECK_AL_ERROR_DEBUG();
-    ALOGVV("Before alSourcei");
-    alSourcei(_alSource, AL_BUFFER, NULL); CHECK_AL_ERROR_DEBUG();
+    ALOGVV("Before alSourceStop, id=%u", _id);
+    CHECK_AL_ERROR_DEBUG();
+    alSourceStop(_alSource);
+    CHECK_AL_ERROR_DEBUG();
+    ALOGVV("Before alSourcei, id=%u", _id);
+
+    // Note: buffers have to be detached before calling 'alDeleteBuffers',
+    // otherwise 'alDeleteBuffers' will cause AL_INVALID_OPERATION.
+    // Even if it's a streaming source, there's no need to screw around with alSourceUnqueueBuffers.
+    alSourcei(_alSource, AL_BUFFER, 0);
```

Comment:
This line fixed the problem, other changes only make our code safer and stronger.